### DISCUSSION TITLE
Return of unfound value?

### DISCUSSION THREAD

**Discussion by ghost** [nlohmann/json#2856] [Q&A] [answered]
Perhaps this is in the exception handling documentation somewhere, but I wasn't able to find a specific answer to this - what happens if a value isn't found? Is there a default that would be returned? For example, I have several functions that I want to call after reading an object out of JSON, but not all of the functions have values that I need to set, they can be zeros or nulls. So if the DOM returns a zero for element["thing"].get<int>() if the element doesn't exist in the JSON, that's fine. Can you elaborate what might happen in this scenario?

**Comment by nlohmann** [comment, marked as answer]
There are multiple ways, see https://json.nlohmann.me/features/element_access/.

**Reply by ghost** [reply]
Thank you, the note about a null value being inserted on empty elements answers it, I think! :)

**Reply by gregmarr** [reply]
@aaronstackpole Just note the difference in behavior between const and non-const versions.
